content/advent-2018/observability-in-the-kitchen.md

```diff
@@ -0,0 +1,204 @@
+GND    | Ground  | P1-06
+SCL    | I2C SCL | P1-05
+SDA    | I2C SDA | P1-03
+
+Then I installed [Raspbian](https://www.raspberrypi.org/downloads/raspbian/) on the Pi and enabled the I2C interface.  I2C is a protocol that allows one device to exchange data with one or more connected devices. 
+
+
+### Prometheus Exporters
+Now I need a way to export metrics in the Prometheus format for the sensor, which means that I need a Prometheus exporter. Exporters are tools that let you translate metrics from other systems into a format that Prometheus can understand. You may be able to find an exporter for your system on the Prometheus [website](https://prometheus.io/docs/instrumenting/exporters/), by searching the Internet, [mailing list](https://groups.google.com/forum/#!forum/prometheus-announce), or looking on GitHub. You can also write your own exporter. That's what I did, and while I did it in Go, it can be done in any programming language.
+ 
+
+### Writing an Exporter in Go
+Here is an overview of how to write a Prometheus exporter for the BME280 sensor in Go.  I need to import the [Prometheus Go client library](https://github.com/prometheus/client_golang) and I will also make use of the [Gobot framework](https://gobot.io/) in order to control the sensor and the Pi.
+
+```
+import (
+	"log"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"gobot.io/x/gobot/drivers/i2c"
+	"gobot.io/x/gobot/platforms/raspi"
+)
+```
+
+Then I create a collector type that can be made aware to the Prometheus client.
+
+```
+type collector struct {
+	sensorDriver *i2c.BME280Driver
+}
+```
+
+In Go your collectors must implement the [prometheus.Collector](https://github.com/prometheus/client_golang/blob/master/prometheus/collector.go) interface. That is to say, the collectors must be objects with the `Describe` and `Collect` methods with a specific signature.
+
+The `Describe` method returns a description of the metrics that it will produce, in particular the metric name, label names, and help string.  This method is used to avoid duplicate registration of metrics and is called at registration time. 
+
+```
+var (
+	temperatureDesc = prometheus.NewDesc("bme280_temperature_celsius", "Temperature in celsius degree", nil, nil)
+	humidityDesc    = prometheus.NewDesc("bme280_humidity_percentage", "Humidity in percentage of relative humidity", nil, nil)
+)
+
+func (c collector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- temperatureDesc
+	ch <- humidityDesc
+}
+```
+
+The `Collect` method fetches all the data you need from the application instance and send the metrics back to the client library. Prometheus client libraries offer four core metric types (counters, gauges, histograms, summaries) but only the gauge is needed for this exporter since the metric needs to represent a value that can either go up or down. These metrics will then be returned by the scrape of the `/metrics` endpoint.
```

Comment:
in `from the application instance and send`: s/send/sends

Reply:
fixed